### PR TITLE
Fix #9413 - allow Redis::connect() to have 7 parameters

### PR DESCRIPTION
**Pull request by thomas-0816**
Fixes https://github.com/phpstan/phpstan/issues/9413

### CODE REVIEW

File: resources/functionMap.php [hunk 1]
```diff
@@ -9262,7 +9262,7 @@
 'Redis::client' => ['mixed', 'command'=>'string', 'arg='=>'string'],
 'Redis::close' => ['bool'],
 'Redis::config' => ['string', 'operation'=>'string', 'key'=>'string', 'value='=>'string'],
-'Redis::connect' => ['bool', 'host'=>'string', 'port='=>'int', 'timeout='=>'float', 'persistent_id='=>'?string', 'retry_interval='=>'?int', 'read_timeout='=>'float'],
+'Redis::connect' => ['bool', 'host'=>'string', 'port='=>'int', 'timeout='=>'float', 'persistent_id='=>'?string', 'retry_interval='=>'?int', 'read_timeout='=>'float', 'context'=>'?array{auth?:array,stream?:array}'],
```
Comment (ondrejmirtes) [inline resources/functionMap.php]: Should be optional - `'context='`

Comment (ondrejmirtes) [reply resources/functionMap.php]: And what kind of arrays are those? Could be described precisely

Comment (thomas-0816) [reply resources/functionMap.php]: seems there are multiple structures possible for auth:
https://github.com/phpredis/phpredis/blob/develop/library.c#L4358
[NULL, 'password']
[false, 'password']
['password']

Comment (thomas-0816) [reply resources/functionMap.php]: for "stream" I guess we can assume array<string, mixed> ?
https://github.com/phpredis/phpredis/blob/develop/library.c#L3257

Comment (ondrejmirtes) [reply resources/functionMap.php]: Yeah sure :)

Comment (thomas-0816) [reply resources/functionMap.php]: I've added a more precise type description for the context elements.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**